### PR TITLE
Features/pragmatic architecture

### PR DESCRIPTION
**Pull request by bart-blommaerts**

### CODE REVIEW

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+Every system has an architecture, even when it is not formally defined. 
+The architecture of a system is typically described in an **Architectural Description**.
+The architectural description documents the system for the stakeholders and needs to make architectural decisions **explicit**.
+The goal of the architectural description is to help in understanding how the system will behave.
+
+<p style="text-align: center;">
+  <img class="image fit" alt="Views and Perspecives" src="/img/prag-arch/views-and-perspectives.png">
+</p>
+
+Following the approach in the book [Software Systems Architecture](http://www.viewpoints-and-perspectives.info/) by [Nick Rozanski](https://twitter.com/nickrozanski) and [Eoin Woods](https://twitter.com/eoinwoodz) an architectural description is composed of a number views.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: **,** an architectural 
(add a comma)

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+* Internationalisation
+* Location
+* Performance and Scalability
+* Regulation
+* Security 
+* Usability
+
+Again, summaries are available on the website of the book.
+If you want a more in-depth explanation, I really recommend reading the book.
+In today's, agile world, I believe the **Evolution perspective** is a key differentiator in any architectural description.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Think you may leave out the comma at the start.
In today's agile world, ...

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+## Example
+
+<p style="text-align: center;">
+  <img class="image fit" alt="Example" src="/img/prag-arch/f1.jpg">
+</p>
+
+This is the 2017 Mercedes F1 W08 Hybrid. 
+It weights 702kg and has between 750 and 850 horsepower. 
+It is made out of approximately 80 000 different components.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Just nitpicking but I'd write it as 80.000. 😛

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+It weights 702kg and has between 750 and 850 horsepower. 
+It is made out of approximately 80 000 different components. 
+The price of the car is an estimated 10 million Eur. 
+That is just for the car, not for the R&amp;D that made the car possible.
+
+Looking back at the viewpoints from above, it is easy to identify how these relate to the construction of the car:
+
+* A Formula One car needs a very specific factory (_Development View_).
+It is not built in the same factory Mercedes builds its trucks.
+* The cars need to be transported all around the world (all the Formula One cars travel over 100 000 miles in the air).
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: More nitpicking: 100.000

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+A lot of the success of the car, must be its architecture then.
+More specifically, its **ability to change**: to adapt to new rules, competitors and market change.
+If the architecture of a system, has the ability to change, it immediately has a competitive advantage.
+This is especially true in agile projects.
+
+> [Grady Booch](https://nl.wikipedia.org/wiki/Grady_Booch)
+>
+> Architecture represents the significant design decisions that shape a system, where significant is measured by cost of change.
+
+Often it is very difficult to get a system right from the beginning.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Often, ...
(comma)

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+That is why creating a system that has the ability to evolve is important.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: I'd include two commas:
That is why creating a system, that has the ability to evolve, is important.

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+> [Russell L. Ackoff](https://en.wikipedia.org/wiki/Russell_L._Ackoff)
+>
+> A system is more than the sum of its parts; it is an indivisible whole. 
+> It loses its essential properties when it is taken apart.
+
+To ensure system integrity, the software architect needs to be part of the project team.
+He must make sure that he enables the right people to collaborate on the system architecture.
+Being part of the team does not mean not taking **responsibility**.
+It is great to give **ownership** to members of the team, but in the end the architect needs to stay accountable.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: ... but in the end, the ...
(comma)

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+</p>
+
+While code is a very important asset of a working system, code alone is not enough to have an accurate and lasting description of a system.
+
+> [Grady Booch](https://nl.wikipedia.org/wiki/Grady_Booch)
+>
+> One cannot see the ocean's currents by studying drops of water.
+
+The goal of visually representing a system, through the architectural description, is to make sure the architecture of the system is in the stakeholder's heads.
+The visual representation can be used to check for consistency,  reporting, validation and **sharing information**.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: consistency, reporting
(two spaces instead of one)

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+
+### Some ground rules
+
+While UML has its merits, often it is not necessary to create an extensive UML model for the architecture.
+It will be time-consuming and, unfortunately, often UML is not correctly understood by stakeholders.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Would rephrase this a bit:
It will be time-consuming and, unfortunately, **it is often the case that** UML is not ...

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+* **Always** provide a legend.
+Explain what a certain line or box means. 
+Don't make stakeholders guess.
+* Don't be afraid to add text to a diagram.
+* Don't model what nobody needs. 
+Eg. if you are not using a data store, do not create an _Information View_.
+* Make sure your stakeholders understand what you are documenting.
+
+Whatever your preferred visualisation approach is, keep a **decision log**.
+Document your decisions, the considered alternatives and the timing a decision was made.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Maybe this sounds better?
... and the timing **when** a decision was made.

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+* Used technology
+* Automation
+* Architectural patterns
+* Layering
+* Evolution
+* ...
+
+> [Simon Brown](http://www.codingthearchitecture.com/authors/sbrown/)
+>
+> Just enough up front design to create firm foundations for the software product and its delivery.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Up-front
(hyphen, to be consistent with the previous usage)

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+All team members need to collaborate to the same end-goal, which requires strong communication skills.
+Including team members in defining the technical vision is useful to make sure everybody knows how they, individually, are contributing to the technical vision on a day-to-day basis. 
+
+### Politics
+
+The architecture of a system will have a large impact on the implementation, delivery and usage of the system.
+Systems generally consist of multiple parts and it is the responsibility of the architect to focus on system integrity, creating a system that has a built-in ability to respond to change.
+When the system lacks integrity, it will rapidly become _a system nobody wants to touch_.
+Unfortunately many enterprises have this fear of change embedded in their culture and it will take strategy and sound people skills to prevent this from happening.
+Influence Maps present an interesting way to map  relationships between people and to visualise who influences who, in an enterprise.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: ... to map relationships ...
(two spaces after "map")

File: _posts/2017-06-21-pragmatic-architecture-today.md [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+Any architectural model introduces abstraction and removes noise.
+This model should be well-understood and feedback loops can help with this.
+As an example, comparing a written down version with bullet points of a certain idea, will help in verifying that the message hasn't changed.
+This insight should be mapped on the model.
+
+* **Observe**: Observing both external and internal circumstances or dependencies of your systems.
+	* Collect up-to-date information from different sources: stakeholders, competitors, similar systems, other viewpoints,...
+* **Orient**: Using your past experience to make sense of these observations.
+	* Analyse the observed information and use it to update your current reality. View events, filtered through your own experiences and perceptions.
+* **Decide**: Deciding on a response, because their might be multiple alternative solutions.
```
Comment (yannickdeturck) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: their -> there

File: _posts/2017-06-21-pragmatic-architecture-today.md (hunk 10 above)
Comment (bart-blommaerts) [inline _posts/2017-06-21-pragmatic-architecture-today.md]: Meh .. :)
